File: python/src/kalourmade_sailup/error_mapping.py
from .exceptions import ApiError, AuthenticationError, RateLimitError, ValidationError


def raise_for_status(status: int, raw_body: str) -> None:
    if status < 400:
        return

    message = f"Sailup API error (status {status})"

    if status == 401:
        raise AuthenticationError(message)
    if status == 429:
        raise RateLimitError(message)
    if status in (400, 422):
        raise ValidationError(message)
    raise ApiError(status, raw_body, message)

File: node/src/errorMapping.ts
import { ApiError, AuthenticationError, RateLimitError, ValidationError } from './errors.js';

export function throwIfError(status: number, rawBody: string): void {
  if (status < 400) {
    return;
  }

  const message = `Sailup API error (status ${status})`;

  if (status === 401) throw new AuthenticationError(message);
  if (status === 429) throw new RateLimitError(message);
  if (status === 400 || status === 422) throw new ValidationError(message);
  throw new ApiError(status, rawBody, message);
}
